packages/app-store/campsite/api/callback.ts

```diff
@@ -0,0 +1,68 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import prisma from "@calcom/prisma";
+
+import getAppKeysFromSlug from "../../_utils/getAppKeysFromSlug";
+import getInstalledAppPath from "../../_utils/getInstalledAppPath";
+import createOAuthAppCredential from "../../_utils/oauth/createOAuthAppCredential";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const userId = req.session?.user.id;
+  if (!userId) {
+    return res.status(404).json({ message: "No user found" });
+  }
+  const { code } = req.query;
+  const { client_id, client_secret } = await getAppKeysFromSlug("campsite");
+
+  const result = await fetch(`https://auth.campsite.co/oauth/token`, {
+    method: "POST",
+    body: JSON.stringify({
+      grant_type: "authorization_code",
+      code,
+      client_id,
+      client_secret,
+      redirect_uri: `${WEBAPP_URL}/api/integrations/campsite/callback`,
+    }),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  if (result.status !== 200) {
+    let errorMessage = "Something is wrong with the Campsite API";
+    try {
+      const responseBody = await result.json();
+      errorMessage = responseBody.error;
+    } catch (e) {}
+
+    res.status(400).json({ message: errorMessage });
+    return;
+  }
+
+  const responseBody = await result.json();
+  if (responseBody.error) {
+    res.status(400).json({ message: responseBody.error });
+    return;
+  }
+
+  /**
+   * With this we take care of no duplicate Campsite keys for a single user
+   * when creating a room using deleteMany if there is already a Campsite key
+   * */
+  await prisma.credential.deleteMany({
+    where: {
+      type: "campsite_conferencing",
+      userId,
+      appId: "campsite",
+    },
+  });
+
+  await createOAuthAppCredential(
+    { appId: "campsite", type: "campsite_conferencing" },
+    { access_token: responseBody.access_token },
+    req
+  );
+
+  res.redirect(getInstalledAppPath({ variant: "conferencing", slug: "campsite" }));
```

Comment:
could you add returnTo here similar to what we have in zoom callback 

https://github.com/calcom/cal.com/blob/92845d3adc702e2cf01a19519bac40af2f3a74df/packages/app-store/zoomvideo/api/callback.ts#L81

Reply:
this would allow campsite to be redirected to the app install flow
![Screenshot 2024-07-10 at 9 22 44 PM](https://github.com/calcom/cal.com/assets/32706411/476f0d81-882f-479f-9055-1cd4e899e5d7)

Reply:
Oh, nice! That's way better. Thanks for the suggestion. Updated in https://github.com/calcom/cal.com/pull/15713/commits/922265f63436d6b9fe6412b90ff1e111f50bb7d3.